xaitk_saliency/impls/perturb_image/square_grid.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+from skimage.transform import resize
+from typing import Optional, Dict, Any
+
+from xaitk_saliency import PerturbImage
+from smqtk_descriptors.utils import parallel_map
+
+
+class RandomSquareGrid (PerturbImage):
+    def __init__(
+        self,
+        n: int,
+        s: int,
+        p: float,
+        seed: Optional[int] = None,
+        threads: Optional[int] = None,
+    ):
+        self.n = n
+        self.s = s
+        self.p = p
+        self.seed = seed
+        self.threads = threads
+
+    def perturb(
+        self,
+        ref_img: np.ndarray,
+    ) -> np.ndarray:
+        num_masks = self.n
+        s = np.array([self.s, self.s])
+        p = self.p
+        threads = self.threads
+        grid_rng = np.random.default_rng(self.seed)
+        shift_rng = np.random.default_rng(self.seed)
+
+        img_shape = np.array(ref_img.shape[:2])
+
+        grid_size = (img_shape // s) + 1
+        grid_size += img_shape % s != 0
```

Comment:
Things seem to work, but this math threw me for a loop as to why the `+1` is here, etc. A comment or two might address why this is being done.